pkg/bmc/ilo5.go

```diff
@@ -0,0 +1,47 @@
+package bmc
+
+import (
+	"net/url"
+)
+
+type iLO5AccessDetails struct {
+	hostname string
+	path     string
+}
+
+const iLO5DefaultScheme = "https"
+
+func init() {
+	registerFactory("ilo5+virtualmedia", newiLO5VirtualMediaAccessDetails)
+}
+
+func newiLO5VirtualMediaAccessDetails(parsedURL *url.URL) (AccessDetails, error) {
+	return &iLO5AccessDetails{
+		hostname: parsedURL.Hostname(),
+		path:     parsedURL.Path,
+	}, nil
+}
+
+func (a *iLO5AccessDetails) Type() string {
+	return "ilo5"
+}
+
+func (a *iLO5AccessDetails) NeedsMAC() bool {
+	return true
+}
+
+func (a *iLO5AccessDetails) Driver() string {
+	return "redfish"
+}
+
+func (a *iLO5AccessDetails) DriverInfo(bmcCreds Credentials) map[string]interface{} {
+	return map[string]interface{}{
+		"ilo_username": bmcCreds.Username,
+		"ilo_password": bmcCreds.Password,
+		"ilo_address":  a.hostname,
```

Comment:
The redfish driver doesn't accept these options: https://docs.openstack.org/ironic/latest/admin/drivers/redfish.html#registering-a-node-with-the-redfish-driver